internal/types/document_test.go

```diff
@@ -144,6 +144,33 @@ func TestDocument(t *testing.T) {
 					)),
 				)),
 			},
+			{
```

Comment:
These unit tests are provided for documentation purposes only (to understand how `SetByPath` should work for complicated array case). Proper testing is done through compat tests (see `TestUpdateFieldCompatSet`).